launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/FileUploadServlet.java

```diff
@@ -0,0 +1,295 @@
+import java.io.FileInputStream;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PropertyContainer;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.auth.appengine.AppEngineCredentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.launchpod.data.RSS;
+import com.google.launchpod.data.UserFeed;
+import com.google.launchpod.data.MP3;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.PostPolicyV4;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.common.base.Strings;
+
+@WebServlet("/create-by-upload")
+@MultipartConfig
+public class FileUploadServlet extends HttpServlet {
+
+  public static final String PROJECT_ID = "launchpod-step18-2020"; // The ID of your GCP project
+  public static final String BUCKET_NAME = "launchpod-mp3-files"; // The ID of the GCS bucket to upload to
+
+  public static final String USER_FEED = "UserFeed";
+  public static final String PODCAST_TITLE = "title";
+  public static final String DESCRIPTION = "description";
+  public static final String LANGUAGE = "language";
+  public static final String EMAIL = "email";
+  public static final String TIMESTAMP = "timestamp";
+  public static final String MP3 = "mp3";
+  public static final String MP3_LINK = "mp3Link";
+  public static final String XML_STRING = "xmlString";
+  public static final String PUB_DATE = "pubDate";
+  public static final String GENERATE_RSS_URL = "https://launchpod-step18-2020.appspot.com/rss-feed?action=generateRSSLink&id=";
+  public static final String HTML_FORM_END = "  <input type='file' name='file'/><br />\n"
+                                             + "<input type='submit' value='Upload File' name='submit'/><br />\n"
+                                             + "</form>\n";
+
+  private static final String ID = "id";
+  private static final String ACTION = "action";
+  private static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("MM/dd/yyyy HH:mm:ss");
+
+
+  private enum Action {
```

Comment:
enums are static constants so each name should be in all caps (e.g. GENERATE_RSS_LINK). also since it's an enum you could use a switch statement below instead of if {} else {}